Removed filtering for Case owned ServiceNow actions from the Management UI and make it usable for Alerts. Added documentation. (#71579)
elastic/kibana: docs/user/alerting/action-types/servicenow.asciidoc

[role="xpack"]
[[servicenow-action-type]]
=== ServiceNow action

> ServiceNow action type using [V2 Table API](https://developer.servicenow.com/app.do#!/rest_api_doc?v=orlando&id=c_TableAPI) to create ServiceNow incidents.

Reviewer: The `V2 Table API` portion didn't translate into a link.

[float]
[[servicenow-connector-configuration]]
==== Connector configuration

ServiceNow connectors have the following configuration properties:

Name::      The name of the connector. The name is used to identify a  connector in the management UI connector listing, or in the connector list when configuring an action.

> URL::    The from address for all emails sent with this connector, specified in `user@host-name` format.

Reviewer: The description here is misleading, I believe a copy / paste mistake?

Author: the same here

Username::  username for 'login' type authentication.

> Password::  password for 'login' type authentication.

Reviewer: I'm not sure if these need the `for 'login' type authentication.` portion.

[float]
[[Preconfigured-servicenow-configuration]]
==== Preconfigured action type

[source,text]
--
 my-servicenow:
   name: preconfigured-servicenow-action-type
   actionTypeId: .servicenow
   config:
     apiUrl: https://dev94428.service-now.com/
   secrets:
     user: testuser
     password: passwordkeystorevalue
--

`config` defines the action type specific to the configuration and contains the following properties:

[cols="2*<"]
|===

| `apiUrl`

> | An address that corresponds to *Sender*.

Reviewer: Is `Sender` part of ServiceNow terminology?

|===

`secrets` defines sensitive information for the action type:

[cols="2*<"]
|===

| `username`
| A string that corresponds to *User*.

| `password`
| A string that corresponds to *Password*. Should be stored in the <<creating-keystore, {kib} keystore>>.

|===

[[servicenow-action-configuration]]
==== Action configuration

ServiceNow actions have the following configuration properties:

> Urgency::              The extent to which the incident's resolution can bear delay.

Reviewer 2: ```suggestion
Urgency::              The extent to which the incident resolution can delay.
```

> Severity::             The severity value of the incident.

Reviewer 2: ```suggestion
Severity::             The severity of the incident.
```

> Impact::               The effect an incident has on business. Can be measured by number of users affected or by the criticality of the business system in question.

Reviewer 2: ```suggestion
Impact::               The effect an incident has on business. Can be measured by the number of affected users or by how critical it is to the business in question.
```

> Short description::    A short text field to describe the incident. The book icon will search the knowledge base using the contents of the short description. 

Reviewer 2: ```suggestion
Short description::    A short description of the incident, used for searching the contents of the knowledge base. 
```

> Description::          The details about the incident issue in question.

Reviewer 2: ```suggestion
Description::          The details about the incident.
```

> Additional comments::  The message text of the email. Markdown format is supported.

Reviewer: The descriptions here are misleading, I believe a copy / paste mistake?

> Additional comments::  The additional instructions or other information meant for the client, and any work notes or steps shoudl be taken to troubleshoot the issue.

Reviewer 2: ```suggestion
Additional comments::  Additional information for the client, such as how to troubleshoot the issue.
```

[[configuring-servicenow]]
==== Configuring ServiceNow

> The email action can send email using many popular SMTP email services.

Reviewer: Same here regarding copy / paste.

> ServiceNow offers free https://developer.servicenow.com/dev.do#!/guides/madrid/now-platform/pdi-guide/obtaining-a-pdi[Personal Developer Instances], which can be used to test creating of the incidents.

Reviewer 2: ```suggestion
ServiceNow offers free https://developer.servicenow.com/dev.do#!/guides/madrid/now-platform/pdi-guide/obtaining-a-pdi[Personal Developer Instances], which you can use to test incidents.
```
